packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1107,6 +1090,86 @@ export class TransactionController extends BaseController<
     return this.nonceTracker.getNonceLock(address);
   }
 
+  private onBootCleanup() {
+    this.initApprovalsOfUnapprovedTransactions();
+    this.loadGasValuesOfUnapprovedTransactions();
+    this.submitApprovedTransactions();
+  }
+
+  /**
+   * Create approvals for all unapproved transactions on current chain.
+   */
+  private initApprovalsOfUnapprovedTransactions() {
+    const chainId = this.getChainId();
+    const unapprovedTransactions = this.state.transactions.filter(
+      (transaction) =>
+        transaction.status === TransactionStatus.unapproved &&
+        transaction.chainId === chainId,
+    );
+
+    for (const transactionMeta of unapprovedTransactions) {
+      this.processApproval(transactionMeta, {
+        shouldShowRequest: false,
+      }).catch((error) => {
+        /* istanbul ignore next */
+        console.error('Error during persisted transaction approval', error);
+      });
+    }
+  }
+
+  /**
+   * Update the gas values of all unapproved transactions on current chain.
+   */
+  private async loadGasValuesOfUnapprovedTransactions() {
+    const chainId = this.getChainId();
+    const isEIP1559Compatible = await this.getEIP1559Compatibility();
+    const unapprovedTransactions = this.state.transactions.filter(
+      (transaction) =>
+        transaction.status === TransactionStatus.unapproved &&
+        transaction.chainId === chainId,
+    );
+
+    for (const transactionMeta of unapprovedTransactions) {
+      try {
+        await updateGasFees({
+          eip1559: isEIP1559Compatible,
+          ethQuery: this.ethQuery,
+          getGasFeeEstimates: this.getGasFeeEstimates.bind(this),
+          txMeta: transactionMeta,
+        });
+        this.updateTransaction(
+          transactionMeta,
+          'TransactionController:loadGasValuesOfUnapprovedTransactions - gas values updated',
+        );
+      } catch (error) {
+        this.failTransaction(transactionMeta, error as Error);
```

Comment:
Rather than cast here, can we just assign the `error` above to `error: Error`?

Reply:
Unfortunately, it doesn't accept the custom type in catch clause. 
`Catch clause variable type annotation must be 'any' or 'unknown' if specified.ts(1196)`